content/articles/geofencing-in-android-with-kotlin/index.md

```diff
@@ -0,0 +1,367 @@
+
+**Check the device location settings and start the Geofence**
+
+```kotlin
+private fun validateGadgetAreaInitiateGeofence(resolve: Boolean = true) {
+        val locationRequest = LocationRequest.create().apply {
+            priority = LocationRequest.PRIORITY_LOW_POWER
+        }
+        val builder = LocationSettingsRequest.Builder().addLocationRequest(locationRequest)
+
+        val client = LocationServices.getSettingsClient(this)
+        val locationResponses =
+            client.checkLocationSettings(builder.build())
+
+        locationResponses.addOnFailureListener { exception ->
+            if (exception is ResolvableApiException && resolve) {
+                try {
+                    exception.startResolutionForResult(
+                        this,
+                        REQUEST_TURN_DEVICE_LOCATION_ON
+                    )
+                } catch (sendEx: IntentSender.SendIntentException) {
+                    Log.d(TAG, "Error getting location settings resolution: ${sendEx.message}")
+                }
+            } else {
+                Toast.makeText(this, "Enable your location", Toast.LENGTH_SHORT).show()
+            }
+        }
+
+        locationResponses.addOnCompleteListener {
+            if (it.isSuccessful) {
+                addGeofence()
+            }
+        }
+    }
+```
+
+Check if the user has chosen to accept or reject the request in the `onActivityResult()` method. If they haven't, re-inquire them.
+
+```kotlin
+override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        validateGadgetAreaInitiateGeofence(false)
+    }
+```
+
+#### Step 5 - Adding and Removing Geofence
+
+**Adding Geofence**
+You'll need a method which `PendingIntent` provides to manage Geofence transitions.
+
+A `PendingIntent` describes both an `intent` and the `action` that should be done in response to it. We'll define a pending intent for a BroadcastReceiver to control the Geofence transitions.
+
+```kotlin
+ private val geofenceIntent: PendingIntent by lazy {
+        val intent = Intent(this, GeofenceBroadcastReceiver::class.java)
+        PendingIntent.getBroadcast(this, 0, intent, PendingIntent.FLAG_UPDATE_CURRENT)
+    }
+```
+
+A GeofencingClient is the most basic way to interact with the geofencing APIs. Create an instance of GeofencingClient. 
+
+```kotlin
+private lateinit var geoClient: GeofencingClient
+```
+
+In the `onCreate()` method, initialize the `geofencingClient`
+
+```kotlin
+geoClient = LocationServices.getGeofencingClient(this)
+```
+
+Also within the onCreate method, add a geofenceList that holds geofences. In this step, we will add one geofence but you can have many geofences. 
+
+```kotlin
+geofenceList.add(Geofence.Builder()
+            .setRequestId("entry.key")
+            .setCircularRegion(latitude,longitude,radius)
+            .setExpirationDuration(Geofence.NEVER_EXPIRE)
```

Comment:
These arguments are also not initialized. Kindly give a sample value for the `latitude`, `longitude`, and `radius`.